src/ansi-c/c_typecheck_initializer.cpp

```diff
@@ -436,9 +436,11 @@ exprt::operandst::const_iterator c_typecheck_baset::do_designated_initializer(
         throw 0;
       }
 
-      assert(index<components.size());
-      assert(components[index].type().id()!=ID_code &&
-             !components[index].get_is_padding());
+      DATA_INVARIANT(index<components.size(),
+                     "Member designator is bounded by components size");
```

Comment:
Coding standard: "Error messages should start with a lower case letter." Applies throughout this PR.